in_toto/gpg/common.py

```diff
@@ -0,0 +1,152 @@
+
+  data = in_toto.gpg.util.parse_packet_header(
+      data, PACKET_TYPES['signature_packet'])
+  ptr = 0
+
+  # we get the version number, which we also expect to be v4, or we bail
+  # FIXME: support v3 type signatures (which I havent' seen in the wild)
+  version_number = data[ptr]
+  ptr += 1
+  if version_number not in SUPPORTED_SIGNATURE_PACKET_VERSIONS: # pragma: no cover
+    raise ValueError("Only version 4 signature packets are supported")
+
+  # here, we want to make sure the signature type is indeed PKCSV1.5 with RSA
+  signature_type = data[ptr]
+  ptr += 1
+  if signature_type != SIGNATURE_TYPE_CANONICAL: # pragma: no cover
+    raise ValueError("We can only use canonical signatures on in-toto")
+
+  signature_algorithm = data[ptr]
+  ptr += 1
+
+  if signature_algorithm not in SUPPORTED_SIGNATURE_ALGORITHMS: # pragma: no cover
+    raise ValueError("This signature algorithm is not supported!")
+
+  key_type = SUPPORTED_SIGNATURE_ALGORITHMS[signature_algorithm]['type']
+  handler = SIGNATURE_HANDLERS[key_type]
+
+  hash_algorithm = data[ptr]
+  ptr += 1
+
+  if hash_algorithm not in SUPPORTED_HASH_ALGORITHMS: # pragma: no cover
+    raise ValueError("This library only supports sha256 as "
+            "the hash algorithm!")
+
+  # obtain the hashed octets.
+  hashed_octet_count = struct.unpack(">H", data[ptr:ptr+2])[0]
+  ptr += 2
+  hashed_subpackets = data[ptr:ptr+hashed_octet_count]
+  hashed_subpacket_info = in_toto.gpg.util.parse_subpackets(hashed_subpackets)
+
+  # check wether we were actually able to read this much hashed octets
+  if len(hashed_subpackets) != hashed_octet_count: # pragma: no cover
+    raise ValueError("this signature packet is missing hashed octets!")
+
+  ptr += hashed_octet_count
+  other_headers_ptr = ptr
+
+  unhashed_octet_count = struct.unpack(">H", data[ptr: ptr + 2])[0]
+  ptr += 2
+  unhashed_subpackets = data[ptr:ptr+unhashed_octet_count]
+  unhashed_subpacket_info = in_toto.gpg.util.parse_subpackets(
+      unhashed_subpackets)
+  ptr += unhashed_octet_count
+
+  # this is a somewhat convoluted way to compute the keyid from the signature
+  # subpackets. Try to obtain the FULL_KEYID_SUBPACKET and bail even if the
+  # partial one is available.
+  keyid = filter(lambda x: True if x[0] == FULL_KEYID_SUBPACKET else False,
+          hashed_subpacket_info)
+
+  # Excluded so that coverage does not vary in different test environments
+  if keyid: # pragma: no cover
+    keyid = binascii.hexlify(keyid[0][1][2:])
+
+  else: # pragma: no cover
+    keyid = ""
+    in_toto.log.warn("can't parse the full keyid on this signature packet."
+        "you need at least gpg version '{}'. Your version is '{}'".format(
+        FULLY_SUPPORTED_MIN_VERSION, in_toto.gpg.util.get_version()))
+
+  left_hash_bits = struct.unpack(">H", data[ptr:ptr+2])[0]
+  ptr += 2
+
+  # Notice the /8 at the end, this length is the bitlength, not the length of
+  # the data in bytes (as len reports it)
```

Comment:
I think this comment is outdated. Are you by any chance referring to [`util.get_mpi_length`](https://github.com/SantiagoTorres/in-toto/blob/41c58355d79af018f1800d5d002479bd8d377f8d/in_toto/gpg/util.py#L49)?

Reply:
yeah, that comment was outdated. I moved it around on dd13b81afe3031ca955ec98c4898c46733b916dc